hudi-common/src/test/java/org/apache/hudi/common/testutils/HoodieAdaptablePayloadDataGenerator.java

```diff
@@ -0,0 +1,228 @@
+    GenericRecord r = new GenericData.Record(SCHEMA);
+    r.put("id", id);
+    r.put("pt", pt);
+    return recordGen.populateForDelete(r, ts);
+  }
+
+  private static HoodieRecord getHoodieRecord(GenericRecord r, Class<?> payloadClass) throws IOException {
+    return new HoodieAvroIndexedRecord(r)
+        .prependMetaFields(
+            SCHEMA,
+            SCHEMA_WITH_METAFIELDS,
+            new MetadataValues().setRecordKey(r.get("id").toString()).setPartitionPath(r.get("pt").toString()),
+            new Properties())
+        .wrapIntoHoodieRecordPayloadWithParams(
+            SCHEMA_WITH_METAFIELDS,
+            getPayloadProps(payloadClass),
+            Option.empty(),
+            false,
+            Option.empty(),
+            false,
+            Option.of(SCHEMA));
+  }
+
+  public static class RecordGen {
+
+    public static final Set<Class<?>> SUPPORTED_PAYLOAD_CLASSES = new HashSet<>(Arrays.asList(
+        OverwriteWithLatestAvroPayload.class,
+        OverwriteNonDefaultsWithLatestAvroPayload.class,
+        PartialUpdateAvroPayload.class,
+        DefaultHoodieRecordPayload.class,
+        AWSDmsAvroPayload.class,
+        MySqlDebeziumAvroPayload.class,
+        PostgresDebeziumAvroPayload.class
+    ));
+
+    private final Class<?> payloadClass;
+    private final String orderingField;
+
+    public RecordGen(Class<?> payloadClass) {
+      checkArgument(SUPPORTED_PAYLOAD_CLASSES.contains(payloadClass));
+      this.payloadClass = payloadClass;
+      if (payloadClass == MySqlDebeziumAvroPayload.class) {
+        orderingField = "_event_seq";
+      } else if (payloadClass == PostgresDebeziumAvroPayload.class) {
+        orderingField = "_event_lsn";
+      } else {
+        orderingField = "ts";
+      }
+    }
+
+    public Class<?> getPayloadClass() {
+      return payloadClass;
+    }
+
+    public String getOrderingField() {
+      return orderingField;
+    }
+
+    GenericRecord populateForInsert(GenericRecord r, long ts) {
+      r.put(orderingField, ts);
+      if (payloadClass == AWSDmsAvroPayload.class) {
+        r.put(AWSDmsAvroPayload.OP_FIELD, "I");
+      }
+      return r;
+    }
+
+    GenericRecord populateForUpdate(GenericRecord r, long ts) {
+      r.put(orderingField, ts);
+      if (payloadClass == AWSDmsAvroPayload.class) {
+        r.put(AWSDmsAvroPayload.OP_FIELD, "U");
+      }
+      return r;
+    }
+
+    GenericRecord populateForDelete(GenericRecord r, long ts) {
+      r.put(orderingField, ts);
+      if (payloadClass == MySqlDebeziumAvroPayload.class) {
+        r.put(DebeziumConstants.FLATTENED_OP_COL_NAME, DebeziumConstants.DELETE_OP);
+      } else if (payloadClass == PostgresDebeziumAvroPayload.class) {
+        r.put(DebeziumConstants.FLATTENED_OP_COL_NAME, DebeziumConstants.DELETE_OP);
+      } else if (payloadClass == AWSDmsAvroPayload.class) {
+        r.put(AWSDmsAvroPayload.OP_FIELD, "D");
+      } else {
```

Comment:
can we also add DefaultHoodieRecordPayload here where we can use a custom maker. 
guess we might need to take in a config may be to optionally leverage delete marker.

Reply:
it's better for a UT for Default payload itself. The `_hoodie_is_deleted` marker is effectively also a custom marker. plus other custom marker like `op` field covered too.